core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAO.scala

```diff
@@ -1,30 +1,79 @@
 import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.rawls.RawlsExceptionWithErrorReport
 import org.broadinstitute.dsde.rawls.config.{AzureConfig, MultiCloudWorkspaceConfig}
 import org.broadinstitute.dsde.rawls.dataaccess.SamDAO
-import org.broadinstitute.dsde.rawls.model.{CreationStatuses, RawlsBillingProject, RawlsBillingProjectName, RawlsBillingProjectResponse, SamResourceAction, SamResourceTypeNames, SamUserResource, UserInfo, AzureManagedAppCoordinates}
+import org.broadinstitute.dsde.rawls.model.{AzureManagedAppCoordinates, CreationStatuses, ErrorReport, RawlsBillingAccountName, RawlsBillingProject, RawlsBillingProjectName, SamResourceAction, SamResourceTypeNames, SamUserResource, UserInfo}
 
+import java.util.UUID
 import scala.concurrent.{ExecutionContext, Future}
+import scala.jdk.CollectionConverters._
 
+/**
+ * Common interface for Billing Profile Manager operations
+ */
 trait BillingProfileManagerDAO {
-  def listBillingProfiles(userInfo: UserInfo, samUserResources: Seq[SamUserResource])(implicit ec: ExecutionContext): Future[Seq[RawlsBillingProject]]
+  def createBillingProfile(displayName: String, billingInfo: Either[RawlsBillingAccountName, AzureManagedAppCoordinates], userInfo: UserInfo): Future[ProfileModel]
+
+  def listBillingProfiles(samUserResources: Seq[SamUserResource], userInfo: UserInfo)(implicit ec: ExecutionContext): Future[Seq[RawlsBillingProject]]
+
+  def listManagedApps(subscriptionId: UUID, userInfo: UserInfo): Future[Seq[AzureManagedAppModel]]
 }
 
 
+class ManagedAppNotFoundException(errorReport: ErrorReport) extends RawlsExceptionWithErrorReport(errorReport)
+
 /**
  * Facade over the billing profile manager service. This service will eventually by the source of truth for
  * billing profiles in the Terra system. For now, we are using this to layer in "external" billing profiles
  * for the purposes of testing Azure workspaces.
  */
-class BillingProfileManagerDAOImpl(samDAO: SamDAO, config: MultiCloudWorkspaceConfig) extends BillingProfileManagerDAO with LazyLogging {
+class BillingProfileManagerDAOImpl(samDAO: SamDAO,
+                                   apiClientProvider: BillingProfileManagerClientProvider,
+                                   config: MultiCloudWorkspaceConfig) extends BillingProfileManagerDAO with LazyLogging {
+
+
+  override def listManagedApps(subscriptionId: UUID, userInfo: UserInfo): Future[Seq[AzureManagedAppModel]] = {
+    val azureApi = apiClientProvider.getAzureApi(userInfo.accessToken.token)
+
+    val result = azureApi.getManagedAppDeployments(subscriptionId).getManagedApps.asScala.toList
+    Future.successful(result)
+  }
+
+  override def createBillingProfile(displayName: String,
+                                    billingInfo: Either[RawlsBillingAccountName, AzureManagedAppCoordinates],
+                                    userInfo: UserInfo): Future[ProfileModel] = {
+    val azureManagedAppCoordinates = billingInfo match {
```

Comment:
@aherbst-broad what are your thoughts about when/how we add test coverage of this method?

Reply:
Good eye, will add some coverage.